Quake/menu.c

```diff
@@ -1354,25 +1354,46 @@ void M_ScanSaves (void)
 	FILE	*f;
 	int	version;
 
+	//load a quicksave
+	strcpy (m_filenames[0], "--- UNUSED SLOT ---");
+	loadable[0] = false;
+	q_snprintf (name, sizeof(name), "%s/quick.sav", com_gamedir);
+	f = Sys_fopen (name, "r");
+	if (f)
+	{
+		fscanf (f, "%i\n", &version);
+		fscanf (f, "%79s\n", name);
+		q_strlcpy (m_filenames[0], name, SAVEGAME_COMMENT_LENGTH+1);
+
+		// change _ back to space
+		for (j = 0; j < SAVEGAME_COMMENT_LENGTH; j++)
+		{
+			if (m_filenames[0][j] == '_')
+				m_filenames[0][j] = ' ';
+		}
+		loadable[0] = true;
+		fclose (f);
+	}
+
 	for (i = 0; i < MAX_SAVEGAMES; i++)
 	{
-		strcpy (m_filenames[i], "--- UNUSED SLOT ---");
-		loadable[i] = false;
+		strcpy (m_filenames[i+1], "--- UNUSED SLOT ---");
```

Comment:
This is certainly broken. The indices must stay the same, but `i` must start at `1` instead of `0`.

Reply:
No, the code inserts quicksave file into slot 0, so manual savefiles have to be shifted by one on the list to match file names, that is quick.sav on slot 0, s0.sav on slot 1, s1.sav on slot 2 and so on

Reply:
Just look at the m_filenames array size: You are writing one more than it can hold when you hit the limit, i.e. you are causing a buffer overrun.

Reply:
I know about this, but I'm unable to fix it right now.
However I also see that all of this could be replaced with
`q_snprintf (name, sizeof(name), i == 0 ? "%s/quick.sav" : "%s/s%i.sav", com_gamedir, i);`

Reply:
> I know about this, but I'm unable to fix it right now.

And yet you create a pull request with a known-broken patch. Geez...

Reply:
I thought about this after creating the pull request. I will fix it.

Reply:
> > I know about this, but I'm unable to fix it right now.
> 
> And yet you create a pull request with a known-broken patch. Geez...

Sorry for causing trouble, bug should be now fixed.